### PR TITLE
fix bug in dotmul_operator's api and anotation

### PR DESCRIPTION
**Pull request by luotao1**

### CODE REVIEW

File: python/paddle/trainer/config_parser.py [hunk 1]
```diff
@@ -2457,13 +2457,11 @@ def __init__(
                 input_layer = self.get_input_layer(input_index)
                 operator_conf.input_sizes.append(input_layer.size)
                 operator_input_index.append(input_index)
-            if self.config.size == 0:
-                size = operator.calc_output_size(operator_conf.input_sizes)
-                if size != 0:
+            size = operator.calc_output_size(operator_conf.input_sizes)
```
Comment (emailweixu) [inline python/paddle/trainer/config_parser.py]: This is not right. It can change size to 0 and make line 2485 fail.

Comment (luotao1) [inline python/paddle/trainer/config_parser.py]: In current version, as line 2461 is the same as line 2465 (`size = operator.calc_output_size(operator_conf.input_sizes)`), thus no matter line 2460 (`if self.config.size == 0`) is true, line 2461 will be executed. So I move this line up front.

Comment (emailweixu) [inline python/paddle/trainer/config_parser.py]: I see. That means the previous one is buggy. In the previous version line 2465 should be changed to 
sz = operator.calc_output_size()
if sz != 0:

Comment (luotao1) [inline python/paddle/trainer/config_parser.py]: have already fix it

File: python/paddle/trainer/config_parser.py [hunk 2]
```diff
@@ -2462,7 +2462,7 @@ def __init__(
                 if size != 0:
                     self.set_layer_size(size)
             else:
-                size = operator.calc_output_size(operator_conf.input_sizes)
+                size = operator.calc_output_size()
```
Comment (emailweixu) [inline python/paddle/trainer/config_parser.py]: sz = operator.calc_output_size(operator_conf.input_sizes)
if sz != 0:
    config_assert(....

**Comment by emailweixu** [comment]
please resovle conflicts

File: python/paddle/trainer_config_helpers/layers.py [hunk 3]
```diff
@@ -407,30 +403,35 @@ def dotmul_projection(input, param_attr=None, scale=1):
...
+    proj.origin = input 
     return proj
 
 def dotmul_operator(x, y, scale=1):
     """
     DotMulOperator takes two inputs and performs element-wise multiplication:
+
     .. math::
-       out.row[i] += scale * (in1.row[i] .* in2.row[i])
+       out.row[i] += scale * (x.row[i] .* y.row[i])
```
Comment (reyoung) [inline python/paddle/trainer_config_helpers/layers.py]: Maybe it need an empty line before to show the correct page.

**Pull request merged**